src/catch2/reporters/catch_reporter_helpers.cpp

```diff
@@ -236,36 +236,50 @@ namespace Catch {
     }
 
     namespace {
-        struct SummaryColumn {
-            SummaryColumn( std::string _label, Colour::Code _colour ):
-                label( CATCH_MOVE( _label ) ), colour( _colour ) {}
-            SummaryColumn addRow( std::uint64_t count ) {
-                ReusableStringStream rss;
-                rss << count;
-                std::string row = rss.str();
-                for ( auto& oldRow : rows ) {
-                    while ( oldRow.size() < row.size() )
-                        oldRow = ' ' + oldRow;
-                    while ( oldRow.size() > row.size() )
-                        row = ' ' + row;
+        class SummaryColumn {
+        public:
+            SummaryColumn( std::string suffix, Colour::Code colour ):
+                m_suffix( CATCH_MOVE( suffix ) ), m_colour( colour ) {}
+
+            SummaryColumn&& addRow( std::uint64_t count ) && {
+                std::string row = std::to_string(count);
+                auto const new_width = std::max( m_width, row.size() );
+                if ( new_width > m_width ) {
+                    for ( auto& oldRow : m_rows ) {
+                        oldRow = std::string( new_width - m_width, ' ' )
+                                     .append( oldRow );
```

Comment:
Tiny nitpick:
```
oldRow.insert(0, new_width - m_width, ' ');
```

is never worse and in _some_ (rather specific) circumstances actually better.

Reply:
Similar change applies to the other branch as well.

Reply:
Fixed and squashed